### DISCUSSION TITLE
Github Sponsors and similar in OpenMage

### DISCUSSION THREAD

**Discussion by Flyingmana** [OpenMage/magento-lts#3495] [General]
Currently, as OpenMage had no legal entity behind it, its not so easy to collect money and sponsorships in the context of the project.

But some of us (me included) already do have Github Sponsors.

My Suggestion, or Question what you think about this:

How about allowing all bigger or regular contributors, to get added to the sponsors.yaml of the OpenMage repository?

Or are there some who would only sponsor to an official legal entity?

**Comment by fballiano** [comment]
I'd agree because it may help for sure, but it will be tricky to decide who's entering/exiting this "get featured for sponsoring" state, IMHO it would be fair to have clear rules about it, probably through an RFC process.

**Reply by Flyingmana** [reply]
I added an RFC proposal https://github.com/OpenMage/rfcs/pull/11

**Comment by addison74** [comment]
OpenMage needs fresh forces, People who appreciate the platform and help improve it. If this can be achieved with money, then I think the best solution is for people who want to be remunerated for their work to benefit from money through Revolut, Patreon, Github and any other legal channels through which money can be transferred. Rules can be established so that everyone is satisfied at the end. As far as I am concerned, I am paid from other sources, while OpenMage helps me provide online services.

Your proposal is fine by adding it to Sponsors, at one point it seemed like @sreichel wanted something like that. After a while we can analyze the results and improve.

**Comment by colinmollenhour** [comment]
I think listing the top contributors and allowing third-parties to choose sponsors would be worth trying. It would be a lot easier than dealing with an actual company and income. However, I think without really calling out to the merchants that are using it that they are using software that needs sponsors it won't go very far.. For example, build in some frequent mildly annoying reminders to the GUI that only go away if the user's installation is whitelisted somehow (IP address, unique key stored in config, etc.). I think this is basically "Pay what you want".. After OM being free for so long it may be hard, but also there is a lot at stake for users and they draw great benefit unlike frivolous things like streaming services and gimmicky apps.

In short - pressure users to become sponsors - obligate OM contributors to record their sponsorships in a system of some sort so that the reminders go away for sponsored installations.

**Comment by kiatng** [comment]
I find GitHub Sponsor limiting. We need a new innovative solution. I asked ChatGPT to help me write the following:

# Project Brief: Open Source Sponsorship Platform on [Hedera](https://hedera.com/)

## Project Overview:
We are seeking angel investors to support our groundbreaking project that leverages web 3 technology, specifically based on Hedera, to create a platform for Open Source Sponsorship. In today's digital landscape, open source software plays a critical role, and our platform aims to ensure the sustainability of these projects by facilitating sponsorship in an innovative and transparent way.

## Platform Features:

1. Low Barrier Onboarding: We prioritize accessibility, making it easy for open source software projects and their maintainers to join our platform. This inclusive approach ensures that projects of all sizes can benefit from sponsorship.

2. Embedded JavaScript (JS):
2.1. User Engagement: For non-sponsored users of the software, we employ embedded JS to emit mildly annoying notifications encouraging them to join the platform as sponsors. We offer various sponsorship programs, including Gold, Silver, Bronze, or one-time donations, allowing users to choose the level of their commitment.

2.2. Sponsor Validation: For sponsored users, we utilize embedded JS to validate their sponsorship status. Additionally, we provide reminders to continue sponsoring if their sponsorship expires, ensuring ongoing support for open source projects.

3. Sponsor Showcase: We prominently display sponsors on the platform, giving them recognition and visibility within the open source community. This not only fosters goodwill but also encourages more individuals and organizations to participate.

4. Transparent Fund Distribution: One of our core principles is transparency. We employ Hedera technology to ensure that funds contributed by sponsors are distributed fairly and directly to the maintainers of open source projects. This eliminates intermediaries and guarantees that resources reach those who need them most.

## Hedera Technology Integration:

Hedera technology plays a pivotal role in our Open Source Sponsorship Platform:

1. Consensus Service: Hedera's Consensus Service provides a robust and tamper-proof mechanism for tracking sponsorship transactions and agreements. This ensures the integrity and immutability of sponsorship records, assuring sponsors that their contributions are being used as intended.

2. Hedera Token Service: We leverage the Hedera Token Service to create and manage digital assets that represent sponsorships and donations. These tokens are used within the platform to track and manage contributions, providing a secure and transparent way to handle funds.

3. Hedera Smart Contracts: Smart contracts on Hedera enable us to automate various aspects of the sponsorship process, such as validating sponsorships, managing reminders, and facilitating fund distribution. This automation reduces operational overhead and ensures a seamless experience for users and maintainers.

4. Hedera Consensus Time: Hedera's consensus time, known for its accuracy, is crucial in timestamping transactions and sponsorship agreements, enhancing the transparency and auditability of our platform.

In conclusion, our Open Source Sponsorship Platform on Hedera is poised to revolutionize the way open source projects receive support. By combining Hedera's advanced technology with innovative features, we aim to ensure the continued growth and sustainability of open source software, making it more accessible and transparent for sponsors and maintainers alike. We invite angel investors to join us in this exciting journey to empower the open source community.

**Reply by kiatng** [reply]
This also solves the security bug/vulnerability reward program. All contributors (maintainers, reviewers, issue reporters, testers, coders, graphic designers, writers, translators, musicians, artists) get a eWallet, which can accept FIAT-equivalent of even fraction of a cent. Sponsorship fund in any amount (small and large) is tokenized and can be distributed according to some rules in smart contracts to contributors. Anyone can contribute to any projects and get rewarded. The Platform can provide a sustainable open source ecosystem. 

Exit strategy: sell it to MS GitHub.

**Reply by colinmollenhour** [reply]
Many-to-many Patreon? I'd be really interested to see if any other OSS project has tried something like this. If not, I'm not sure we have the chops on a volunteer basis to be a trailblazer in this area? It is a very interesting concept and probably worthy of it's own company!

The key with GH Sponsors (or Patreon, etc) is it is super easy for the Sponsor to setup recurring payment with a credit card (low friction) and GH handles things like US tax forms for the Sponsoree (Patreon and others too, I'm sure). I don't know anything about how different tax jurisdictions would look upon a 100% crypto-based pseudo-company.. Would the proceeds be considered a gift, or capital gains or regular income? Who would be the payor? How would the payor obtain the crypto and how would they expense it? In the US I believe even exchanges between cryptos are taxed, it is highly regulated, basically the same as fiat. A donation is basically an act of charity or goodwill, but as soon as you start assigning amounts based on work efforts it starts to look more like a company paying "employees".

**Reply by Flyingmana** [reply]
there is in theory https://opencollective.com/ although Iam not sure if its usable without a legal entity.
Its used by a few bigger and smaller open source and software projects.

I think it might be a better alternative then Hedera, as any "coin" related approach is legally tricky and might put you fast on money laundry watch and block lists

**Reply by colinmollenhour** [reply]
OpenCollective looks very interesting! What do you think about build in a "feature" that displays a "please donate" bar that goes away for say 6 months after the site is matched to a donation? Then the question is how to divvy up disbursements but it shouldn't be too hard to devise something that is better than everybody getting $0. 🤣

**Reply by kiatng** [reply]
We don't need to be a legal entity if we can find a host:

[What is a fiscal host?](https://opencollective.com/fiscal-hosting)
A fiscal host is an organization that welcomes others to operate through their structure, so projects can use the host’s legal entity and bank account instead of setting up their own. The host provides administrative services, oversight, and support.

*Fiscal hosting is also called fiscal sponsorship, fund-holding, or auspicing in different places around the world.

There are a list of legal hosts to choose [from](https://opencollective.com/search?isHost=true), which charge various host fees.

**Reply by Flyingmana** [reply]
we have now an OpenCollective account created
https://opencollective.com/openmage

to have a way to set goals, I started to make some rough plans how to make use of money we would get in the future.
https://github.com/OpenMage/organizational/blob/main/finance/expense_planning.md

**Reply by Flyingmana** [reply]
one next step is now linking the opencollective from the repository https://github.com/OpenMage/magento-lts/pull/3649 which requires approvals now
